src/models/Flashcard.js

```diff
@@ -0,0 +1,32 @@
+const mongoose = require('mongoose');
+
+const flashcardSchema = new mongoose.Schema({
+    topic: {
+        type: String,
+        unique: [true, 'This topic is already taken'],
+        required: [true, 'Cannot be blank'],
+        maxlength: 150,
+    },
+    author: {
+        type: String,
+        required: [true, 'Cannot be blank'],
+        maxlength: 100,
+    },
+    question: {
+        type: String,
+        required: [true, 'Cannot be blank'],
+        maxlength: 500,
+    },
+    answer: {
+        type: String,
+        required: [true, 'Cannot be blank'],
```

Comment:
[Loose Schema]
Generally on text fields like this you want to set a maxlength. Otherwise a user with bad intentions (or just making a mistake) could come in and submit things with 10GB of data. A few of those and they'd blow up your database. 😢

Reply:
Potentially it makes you susceptible to a DDOS attack. (Distributed) Denial of Service.
Its where a hacker can make your service unavailable for all users by pushing something to it that is too expensive to handle bringing the overall system down. Thus denying service to your users. In this case, they'd fill up your database and crash the system.

Mongoose isn't the best place actually to fix this. Generally, you'd want this limitation on Mongoose, but also you'd want a check earlier in the process to prevent someone from sending something so large to you.

Reply:
Understood! I added maxlength to the answer field.